app/controllers/proposals_controller.rb

```diff
@@ -1,33 +1,33 @@
 class ProposalsController < ApplicationController
   include ApplicationHelper
   before_action :authenticate_user!, :except => [:show, :index, :related_vote_in_tree, :related_proposals]
-  before_action :find_hub, :find_user, only: :index
 
   # GET /proposals.json
   def index
```

Comment:
Having this many different parameters and scopes in the same method is a code smell - you can define a few different index methods accessed by different routes (e.g. `/hubs/:hub/proposals`, `/users/:id/proposals`, `/proposals/mine`) that use the same base method and make this a lot cleaner. I avoided doing that here because my goal was to clean up the code base without changing the exposed API.